Create Events Page: Split the toggle buttons into 2 sections, and make one of 3 toggles required

JohnCox2211 opened the pull request:
The Bonners toggle will automatically toggle for Bonner events, but can be changed by the user if they wish and other event types can also use this toggle. The toggle state is saved between create event and edit event. The toggles are divided as requested into required and optional categories as well.

JohnCox2211 committed:
added headers for toggle section of create event
page. Added dividing line and styling in
new file createEvent.css. Need to ask if we want
toggle to check if event is Bonners

JohnCox2211 committed:
added additional css formatting, added bonners
toggle. laid out click handler for toggle
switches to implement tomorrow

JohnCox2211 committed:
You may only toggle one event in the must select
one category in create event page now. Fixed
formatting for when the webpage is in a smaller
window.

JohnCox2211 committed:
Shortened form submission toggle check code.

JohnCox2211 committed:
fixed code improperly displaying location on
event edit page.

JohnCox2211 committed:
added event class isbonners boolean field

JohnCox2211 committed:
reformatted toggle headers to stay with their
groups when the page is small

JohnCox2211 committed:
the is bonners toggle will now auto toggle when
the user selectes to create a bonner event.

JohnCox2211 committed:
working prototype. You can have a bonners event
that is not checked as bonners, but the user
has freedom to check other boxes if wanted

JohnCox2211 committed:
toggles will now behave correctly between pages.

JohnCox2211 commented:
https://github.com/BCStudentSoftwareDevTeam/celts/issues/1294 - github issue

https://trello.com/c/50TXOdHE - trello card

JohnCox2211 committed:
Fixed grammar (changed bonners to bonner), view
event tab now reflects whether the event is
bonners or not

BrianRamsay reviewed (changes requested):
The bonner toggle only should be there for Bonner events.

Let's clean up the html classes and styles and try and simplify what controls the width of the components. Use the bootstrap grid only, if possible, for rows and columns.

JohnCox2211 committed:
reformatting toggle switch code to allow only one
selected, but if a new toggle is selected, the
others are untoggled and the click one is on. Gave
the divider line its own div, but it sill needs
formatting work.

JohnCox2211 committed:
Divide line is now formatted to disappear once the
window is below a certain size

JohnCox2211 committed:
fixed line width, ready for new PR

BrianRamsay commented on app/static/js/createEvents.js:
```diff
@@ -98,10 +104,25 @@ $(document).ready(function () {
+  // Determine which checkbox was clicked and its current checked status, uncheck others
+  $("#checkIsTraining, #checkServiceHours, #checkBonners").on('click', function (event) {
+    let $target = $(event.target);
+    $("#checkIsTraining, #checkServiceHours, #checkBonners").not($target).prop('checked', false);
```
we can just use `$(event.target)` in the `not()`

BrianRamsay reviewed (changes requested):
Let's clean up the css so we are relying more on bootstrap. the cutoff point should be xl, then you don't need min-width or the @media selector

BrianRamsay committed:
Merge branch 'development' into eventtoggles

JohnCox2211 committed:
Made specified changes: col size now xl. Removed
media sizing in CSS, removed JS variable and
added its code directly to the line it was used
in.
